domain/user/state/state.go

```diff
@@ -291,6 +291,86 @@ func (st *State) SetPasswordHash(ctx context.Context, uuid user.UUID, passwordHa
 	})
 }
 
+// EnableUser will enable the user with the supplied UUID. If the user does not
+// exist an error that satisfies usererrors.NotFound will be returned. If the
+// user is already enabled an error that satisfies usererrors.AlreadyEnabled
+// will be returned.
+func (st *State) EnableUser(ctx context.Context, uuid user.UUID) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Annotate(err, "getting DB access")
+	}
+
+	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
```

Comment:
We probably need to ensure that the user has a user_authentication record before running the update. Might be better to turn this query into an upsert to make sure it is there and set. You get it all in one operation then.

Reply:
Need your review of how I handle the `ensureUserAuthentication` errors for `EnableUserAuthentication`, it has some specifics.